nf_core/components/components_utils.py

```diff
@@ -161,48 +162,35 @@ def get_components_to_install(
-                        "org_path": None,
-                        "git_remote": None,
-                        "branch": None,
+                    component_name = "/".join(name_split)
+                    component_dict: Dict[str, Optional[str]] = {
+                        "name": component_name,
                     }
-                    modules.append(component_dict)
+                    modules[component_name] = component_dict
                 elif link.startswith("../"):
-                    component_dict = {"name": name.lower(), "org_path": None, "git_remote": None, "branch": None}
-                    subworkflows.append(component_dict)
+                    component_name = name.lower()
+                    component_dict = {"name": component_name}
+                    subworkflows[component_name] = component_dict
 
     if Path(subworkflow_dir, "meta.yml").exists():
         with open(Path(subworkflow_dir, "meta.yml")) as fh:
             meta = yaml.safe_load(fh)
-            if "components" not in meta:
-                return modules, subworkflows
-            components = meta.get("components")
-            new_module_list = []
-            new_subwf_list = []
-
-            for component in components:
-                if isinstance(component, str):
-                    component_dict = {"name": component, "org_path": None, "git_remote": None, "branch": None}
-                else:
-                    name = list(component.keys())[0]
-                    git_remote = component[name]["git_remote"]
-                    org_path_match = re.search(r"(?:https://|git@)[\w\.]+[:/](.*?)/", git_remote)
-                    org_path = org_path_match.group(1) if org_path_match else None
-                    component_dict = {
-                        "name": name.lower(),
-                        "org_path": org_path,
-                        "git_remote": git_remote,
-                        "branch": component[name].get("branch", "master"),
-                    }
-
-                if component_dict["name"] in [sw["name"] for sw in subworkflows]:
-                    new_subwf_list.append(component_dict)
-                else:
-                    if component_dict["name"] in [m["name"] for m in modules]:
-                        new_module_list.append(component_dict)
-
-        modules = new_module_list
-        subworkflows = new_subwf_list
-
-    return modules, subworkflows
+            if "components" in meta:
+                components = meta.get("components")
+                for component in components:
+                    if isinstance(component, dict):
+                        component_name = list(component.keys())[0].lower()
+                        git_remote = component[component_name]["git_remote"]
+                        org_path_match = re.search(r"(?:https://|git@)[\w\.]+[:/](.*?)/", git_remote)
+                        org_path = org_path_match.group(1) if org_path_match else None
```

Comment:
Isn't `org_path` mandatory for the `modules.json` ? I wonder if you should raise an error if the regular expression can't find the org path

Reply:
That does make sense, if a remote different from nf-core is defined it should always be able to get the org_path, or else the rest of the code won't function.

What do you think about this:
```suggestion
                        if org_path_match:
                            org_path = org_path_match.group(1)
                        else:
                            raise UserWarning(f"Organisation path could not be established for {component_name}") 
```

Reply:
"Warning" doesn't quite convey the scale of the issue. I think an exception like `ValueError` or `SyntaxError` would be clearer.

Reply:
I based it on https://github.com/sanger-tol/nf-core-tools/blob/3c493b9f8fc941b86a544e70251aa9e4f5c8311e/nf_core/components/components_utils.py#L86 a few lines above.

But it can also be:

```suggestion
                        if org_path_match:
                            org_path = org_path_match.group(1)
                        else:
                            raise ValueError(f"Organisation path could not be established for {component_name}") 
```

Reply:
Oki. In that case, let's stick with `UserWarning`. Better to be consistent within the repository